### PR TITLE
list users orders

### PR DESCRIPTION
**Pull request by pkr5**

### CODE REVIEW

File: src/main/java/com/project/stockexchangeappbackend/rest/UsersController.java [hunk 1]
```diff
@@ -196,4 +199,16 @@ public String getUsers() {
                 .map(transaction -> mapper.map(transaction, TransactionDTO.class));
     }
 
+    @GetMapping("/{id}/order")
+    @PreAuthorize("hasRole('ADMIN')")
+    @ApiOperation(value = "List user's orders", response = StockDTO.class,
```
Comment (pkr5) [inline src/main/java/com/project/stockexchangeappbackend/rest/UsersController.java]: widzę to, czekam na review i poprawię

Comment (karolcymerys) [reply src/main/java/com/project/stockexchangeappbackend/rest/UsersController.java]: ok

File: src/main/java/com/project/stockexchangeappbackend/rest/UsersController.java [hunk 2]
```diff
@@ -196,4 +199,16 @@ public String getUsers() {
...
+            notes = "Required role: ADMIN")
+    @ApiResponses(@ApiResponse(code = 200, message = "Successfully listed user's orders"))
+    public List<OrderDTO> getUserOwnedOrders(@ApiParam(value = "The users's id.", required = true) @PathVariable Long id) {
```
Comment (karolcymerys) [inline src/main/java/com/project/stockexchangeappbackend/rest/UsersController.java]: You should return Page,i guess

Comment (karolcymerys) [reply src/main/java/com/project/stockexchangeappbackend/rest/UsersController.java]: Ans Specification would be useful, and Pageable as well

File: src/main/java/com/project/stockexchangeappbackend/service/OrderServiceImpl.java [hunk 3]
```diff
@@ -85,6 +85,14 @@ public void deactivateOrder(Long id) {
         archivedOrderRepository.save(archivedOrder);
     }
 
+    @Override
+    @LogicBusinessMeasureTime
+    @Transactional(readOnly = true)
+    public List<Order> getOrdersByUser(Long id) {
+        User user = userRepository.getOne(id);
```
Comment (karolcymerys) [inline src/main/java/com/project/stockexchangeappbackend/service/OrderServiceImpl.java]: You should throw EntityNotFound Exception when user doesn't exist

File: src/main/java/com/project/stockexchangeappbackend/rest/UsersController.java (hunk 1 above)
Comment (karolcymerys) [reply src/main/java/com/project/stockexchangeappbackend/rest/UsersController.java]: By the way, How did you know changes will be required?

**Review by karolcymerys — CHANGES_REQUESTED** [review]

**Review by karolcymerys — APPROVED** [review]

**Pull request merged**